GANDLF/utils/imaging.py

```diff
@@ -1,67 +1,79 @@
-from typing import Union
+from typing import Optional, Union
 import os, pathlib, math, copy
+from enum import Enum
 import numpy as np
 import SimpleITK as sitk
 import torchio
+import cv2
 
 from .generic import get_filename_extension_sanitized
 
 
 def resample_image(
-    img, spacing, size=None, interpolator=sitk.sitkLinear, outsideValue=0
-):
+    input_image: sitk.Image,
+    spacing: Union[np.ndarray, list, tuple],
+    size: Optional[Union[np.ndarray, list, tuple]] = None,
```

Comment:
List and Tuple types - minor thing (or if there are many possible types in there or Tuple sizes maybe let's skip it

Reply:
Fixed.